### PR TITLE
ENH: Adding SWIG executable for use in downstream projects (and/or ex…

### PR DESCRIPTION
**Pull request by adamrankin**
…tensions)

### CODE REVIEW

**Comment by adamrankin** [comment]
@Slicer/slicer-core Is anyone available to review?

File: SuperBuild/External_Swig.cmake [hunk 1]
```diff
@@ -68,6 +68,8 @@ if(NOT SWIG_DIR AND NOT Slicer_USE_SYSTEM_${proj})
 "#------------------------------------------------------------------------------
 # Added by '${CMAKE_CURRENT_LIST_FILE}'
 
+mark_as_superbuild(SWIG_EXECUTABLE)
```
Comment (jcfr) [inline SuperBuild/External_Swig.cmake]: This shouldn't be added to the `_env_script`, it should instead be added at the end of the file. See [here](https://github.com/Slicer/Slicer/blob/1b7888e90e233729e4beb4941e8b33e622373655/SuperBuild/External_VTK.cmake#L260) for an example

Comment (jcfr) [reply SuperBuild/External_Swig.cmake]: This should probably be `Slicer_SWIG_EXECUTABLE`

Comment (adamrankin) [reply SuperBuild/External_Swig.cmake]: The variable Slicer_SWIG_EXECUTABLE doens't exist at this point, should I
```cmake
set(Slicer_SWIG_EXECUTABLE ${SWIG_EXECUTABLE})
```
?

**Comment by jcfr** [comment]
@adamrankin I updated the topic, could you quickly try if it works for you ?

As a side note, Swig is currently built only if SimpleITK is enabled. Considering we will most likely move SimpleITK as an extension in the next few months (see [roadmap](https://www.slicer.org/wiki/Documentation/Labs/Slicer5-roadmap#Overall_Goals)), we may have to revisit the approach to ensure your extension can build.

**Comment by adamrankin** [comment]
Yup, new approach works. When SimpleITK gets taken out, I'll copy `external_swig.cmake`

**Comment by jcfr** [comment]
Closing. Fixed in [r28445](http://viewvc.slicer.org/viewvc.cgi/Slicer4?view=revision&revision=28445)

**Pull request closed without merging**